app/smc/pytest/e2e_smoke.py

```diff
@@ -583,3 +599,59 @@ def test_aiclk(arc_chip_dut, asic_id):
         aiclk = arc_chip.arc_msg(ARC_MSG_TYPE_GET_AICLK)[0]
         assert aiclk == clk, f"Failed to set clock to {clk} MHz"
         logger.info(f"AICLK set to {aiclk} MHz successfully")
+
+
+def upgrade_from_version_test(tmp_path: Path, board_name, unlaunched_dut, version):
+    if board_name is None:
+        pytest.skip("Upgrade test requires --board set, skipping upgrade test")
+
+    RECOVERY_URL = "https://github.com/tenstorrent/tt-zephyr-platforms/releases/download/v18.12.0-rc1/fw_pack-18.12.0-rc1-recovery.tar.gz"
+
+    tar_recovery = tmp_path / "recovery.tar.gz"
+    urlretrieve(RECOVERY_URL, tar_recovery)
+    recovery_py = str(
+        Path(__file__).parents[3]
+        / "scripts/tooling/blackhole_recovery/recover-blackhole.py"
+    )
+    # versions we want to check upgrading from
+
+    # flash recovery first
+    try:
+        subprocess.check_call(
+            [
+                "python3",
+                recovery_py,
+                tar_recovery,
+                board_name,
+                "--force",
+                "--select-first",
+            ]
+        )
+    except subprocess.CalledProcessError as e:
+        logger.error(f"blackhole_recovery.py failed with error: {e}")
+        assert False
```

Comment:
Might also log process output as an info or debug output here so we can confirm it worked in CI runs